tests/hiking_test.cpp

```diff
@@ -0,0 +1,196 @@
+#include "catch/catch.hpp"
+
+#include "avatar.h"
+#include "avatar_action.h"
+#include "calendar.h"
+#include "game.h"
+#include "map.h"
+#include "map_helpers.h"
+#include "player_helpers.h"
+#include "point.h"
+
+static void remain_centered( const tripoint &center, avatar &dummy )
+{
+    if( dummy.pos().x < SEEX * int( MAPSIZE / 2 ) ||
+        dummy.pos().y < SEEY * int( MAPSIZE / 2 ) ||
+        dummy.pos().x >= SEEX * ( 1 + int( MAPSIZE / 2 ) ) ||
+        dummy.pos().y >= SEEY * ( 1 + int( MAPSIZE / 2 ) ) ) {
+        dummy.setpos( center );
+        // Verify that only the player is present.
+        REQUIRE( g->num_creatures() == 1 );
+    }
+}
+
+static int distance_test( avatar &dummy, const time_duration &time_spent )
+{
+    clear_map();
+    dummy.set_stored_kcal( dummy.get_healthy_kcal() );
+    dummy.stomach.empty();
+    dummy.guts.empty();
+    dummy.moves = dummy.get_speed();
+    const tripoint center{ 65, 65, 0 };
+    const time_point start = calendar::turn;
+    int calories_spent = 0;
+    int tiles_moved = 0;
+    while( start + time_spent > calendar::turn ) {
+        const int calories_before = dummy.get_stored_kcal();
+        avatar_action::move( dummy, g->m, dummy.pos() + tripoint_east );
+        ++tiles_moved;
+        remain_centered( center, dummy );
+
+        dummy.update_body();
+        dummy.process_turn();
+        calendar::turn += 1_turns;
+        calories_spent += calories_before - dummy.get_stored_kcal();
+    }
+    printf( "player weight: %.2f kg, bmr: %d\n", units::to_kilogram( dummy.get_weight() ),
+            dummy.get_bmr() );
+    return calories_spent;
+}
+
+static bool between( const int num, const int lower, const int high )
+{
+    return num < high && num > lower;
+}
+
+TEST_CASE( "naked_walk", "[hike]" )
+{
+    clear_player();
+    avatar &dummy = g->u;
+    const int cal_spent = distance_test( dummy, 1_hours );
+    CAPTURE( cal_spent );
+    CHECK( between( cal_spent, 204, 234 ) );
+}
+
+TEST_CASE( "naked_run", "[hike]" )
+{
+    clear_player();
+    avatar &dummy = g->u;
+    dummy.toggle_run_mode();
+    const int cal_spent = distance_test( dummy, 1_hours );
+    CAPTURE( cal_spent );
+    CHECK( between( cal_spent, 1032, 1200 ) );
+}
+
+TEST_CASE( "walk_carry", "[hike]" )
+{
+    SECTION( "10 kg" ) {
+        clear_player();
+        avatar &dummy = g->u;
+        dummy.reset_move_mode();
+        item &backpack = dummy.i_add( item( "backpack" ) );
+        dummy.wear( backpack, false );
+        dummy.i_add( item( "thread", calendar::turn, 9367 ) );
+        const int cal_spent = distance_test( dummy, 1_hours );
+        CAPTURE( cal_spent );
+        CHECK( between( cal_spent, 230, 260 ) );
+    }
```

Comment:
This is a perfect candidate for extraction to a helper function.
void test_walk_carry( int carry_weight, int min_cal, int max_cal )